frontend/app/src/utils/assets.ts

```diff
@@ -51,10 +51,32 @@ function score(keyword: string, { name, symbol }: ManagedAsset): number {
 
 export function compareSymbols(a: string, b: string, keyword: string) {
   const search = keyword.toLocaleLowerCase().trim();
-  return (
-    levenshtein(search, a.toLocaleLowerCase()) -
-    levenshtein(search, b.toLocaleLowerCase())
-  );
+  const keywordA = a.toLocaleLowerCase().trim();
+  const keywordB = b.toLocaleLowerCase().trim();
+
+  let rankA = levenshtein(search, keywordA);
+  let rankB = levenshtein(search, keywordB);
+
+  const keywordAHaystackIndex = keywordA.indexOf(search);
+  const keywordANeedleIndex = search.indexOf(keywordA);
+  const keywordBHaystackIndex = keywordB.indexOf(search);
+  const keywordBNeedleIndex = search.indexOf(keywordB);
+
+  const length = search.length;
+
+  if (keywordAHaystackIndex === 0 || keywordANeedleIndex === 0) {
+    rankA -= length + 1;
+  } else if (keywordAHaystackIndex > 0 || keywordANeedleIndex > 0) {
+    rankA -= length;
+  }
+
+  if (keywordBHaystackIndex === 0 || keywordBNeedleIndex === 0) {
+    rankB -= length + 1;
+  } else if (keywordBHaystackIndex > 0 || keywordBNeedleIndex > 0) {
+    rankB -= length;
+  }
+
```

Comment:
Can you provide an example of what this code is supposed to do?

Is it about changing the order position based on the index of the keyword in each symbol?